src/classes/frSetupController.cls

```diff
@@ -157,21 +260,23 @@ public with sharing class frSetupController {
 		List<frMapping__c> upsertList = new List<frMapping__c>();
 		List<frMapping__c> allMappings = new List<frMapping__c>(donationMappings);
 		allMappings.addAll(donorMappings);
+		allMappings.addAll(accMappings);
+		allMappings.addAll(giftTransactionMappings);
+		allMappings.addAll(giftCommitmentMappings);
 		for(frMapping__c mapping : allMappings) {
+
 			if(String.isNotBlank(mapping.sf_Name__c)) {
-                mapping.Name = (mapping.Type__c == DONATION_TYPE ? 'o.' : 'c.') + mapping.sf_Name__c;
+				if(mapping.Type__c == DONATION_TYPE) mapping.Name = 'o.' + mapping.sf_Name__c;
+				if(mapping.Type__c == DONOR_TYPE) mapping.Name = 'c.' + mapping.sf_Name__c;
+				if(mapping.Type__c == PERSON_ACCOUNT_TYPE) mapping.Name = 'a.' + mapping.sf_Name__c;
+				if(mapping.Type__c == GIFT_TRANSACTION_TYPE) mapping.Name = 't.' + mapping.sf_Name__c;
+				if(mapping.Type__c == GIFT_COMMITMENT_TYPE) mapping.Name = 's.' + mapping.sf_Name__c;
 				if(mapping.Name.length() > customSettingNameLength) {
 					mapping.Name = mapping.Name.subString(0, customSettingNameLength);
 				}
-				upsertList.add(mapping);
+				if(!upsertList.contains(mapping)) upsertList.add(mapping);
```

Comment:
Can you explain why this was added?  It seems like this is trying to prevent a possible duplicated mapping, but I don't think this is the right way to handle that edge case as it will just silently remove a mapping that the user thinks will be saved.  I think this should be removed or if there is an error that is requiring it, can you share that here?

Reply:
Removed for now, but did notice that an error could occur here if a user tried to insert a mapping record with a duplicate sf field.